### PR TITLE
[GRANT-236] Download Excel

### PR DESCRIPTION
**Pull request by leon-fw**
- Download Raw Data button on the applications page
- Downloads excel file of COMPLETED workshops
- Displays requested fields. 

image

image

### CODE REVIEW

File: client/constants/request-method.ts [hunk 1]
```diff
@@ -23,4 +23,5 @@ export const API_ENDPOINT = {
   updateWorksopScores: (applicationId: string, scoreId: string) =>
     `/applications/${applicationId}/workshop/${scoreId}`,
   downloadApplicationScore: (applicationId: string) => `/applications/${applicationId}/download`,
+  RAW_DATA: '/applications/raw-data',
```
Comment (karthikktamilmani-freshworks) [inline client/constants/request-method.ts:26]: APPLICATIONS_RAW_DATA!

File: server/src/score/ro/raw-data.ro.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+export class RawDataRo {
+  result: any;
+  constructor(data: WorkshopScore[]) {
+    this.result = [{ ...roHeaders, ...this.convertWorkshopScoreToContent(data) }];
+  }
+  convertWorkshopScoreToContent(data: WorkshopScore[]) {
+    const content = data.map((item) => {
+      return {
+        facilityName: item.application.facilityName,
```
Comment (karthikktamilmani-freshworks) [inline server/src/score/ro/raw-data.ro.ts:23]: Refer application-score.ro.ts -> Facility name and project title should be retrieved from the submission!

File: server/src/score/ro/raw-data.ro.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        projectTitle: item.application.projectTitle,
+        totalEstimatedCost: item.application.totalEstimatedCost,
+        asks: item.application.asks,
+        finalScore: item.finalScore,
+        scoreRatio: `${(item.finalScore / 112).toFixed(3)}`,
```
Comment (karthikktamilmani-freshworks) [inline server/src/score/ro/raw-data.ro.ts:28]: Refer application-score.ro.ts on the value 112! It's different for different forms!

File: server/src/score/ro/raw-data.ro.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  ],
+};
+
+export class RawDataRo {
+  result: any;
+  constructor(data: WorkshopScore[]) {
+    this.result = [{ ...roHeaders, ...this.convertWorkshopScoreToContent(data) }];
+  }
+  convertWorkshopScoreToContent(data: WorkshopScore[]) {
+    const content = data.map((item) => {
```
Comment (karthikktamilmani-freshworks) [inline server/src/score/ro/raw-data.ro.ts:21]: item: WorkshopScore -> this can be typed!

File: server/src/score/ro/raw-data.ro.ts (hunk 3 above)
Comment (leon-fw) [reply server/src/score/ro/raw-data.ro.ts:28]: Oh, I had and old version of the file, no wonder. Thank you, will rebase and fix this

File: client/services/useDownloadXlsx.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { API_ENDPOINT, REQUEST_METHOD } from '@constants';
+import xlsx from 'json-as-xlsx';
+import { useHttp } from './useHttp';
+
+const settings = {
+  fileName: 'MySpreadsheet', // Name of the resulting spreadsheet
```
Comment (karthikktamilmani-freshworks) [inline client/services/useDownloadXlsx.ts:6]: RawData

**Review by karthikktamilmani-freshworks — APPROVED** [review]
One minor comment! Looks good otherwise!

File: client/services/useDownloadXlsx.ts (hunk 5 above)
Comment (leon-fw) [reply client/services/useDownloadXlsx.ts:6]: Overlooked that, thank you!

**Pull request merged**